### PR TITLE
Allow rlang-style lambda expressions in stat_summary functions

### LINKED ISSUE #3568: stat_summary and rlang lambda expressions
While lambda expressions are allowed in `stat_function()` and a few other places, they aren't in `stat_summary()` type functions. Example below. Note that the args `fun.y`, `fun.ymin`, and `fun.ymax` are in the dev version `fun`, `fun.min`, and `fun.max`.

``` r
library("ggplot2")

df <- data.frame(
  x = c(1, 1, 2, 2, 3, 3),
  y = c(0, 2, 1, 3, 2, 4)
)

ggplot(df, aes(x, y)) +
  stat_summary(
    fun.y = mean,
    fun.ymin = function(y) mean(y) - 1,
    fun.ymax = function(y) mean(y) + 1
  )
```

``` r

ggplot(df, aes(x, y)) +
  stat_summary(
    fun.y = mean,
    fun.ymin = ~ mean(.x) - 1,
    fun.ymax = ~ mean(.x) + 1
  )
#> Warning: Computation failed in `stat_summary()`:
#> 'what' must be a function or character string
```

<sup>Created on 2019-10-14 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

### PR DESCRIPTION
**Pull request by dkahle**

### CODE REVIEW

File: R/stat-summary-bin.R [hunk 1]
```diff
@@ -96,7 +96,11 @@ make_summary_fun <- function(fun.data, fun, fun.max, fun.min, fun.args) {
 
   if (!is.null(fun.data)) {
     # Function that takes complete data frame as input
-    fun.data <- match.fun(fun.data)
+    if (is.formula(fun.data)) {
+      fun.data <- rlang::as_function(fun.data)
+    } else {
+      fun.data <- match.fun(fun.data)
+    }
```
Comment (thomasp85) [inline R/stat-summary-bin.R]: I think you can just use `as_function()` for everything as it behaves correctly in all cases... We also import rlang so no need to prefix it

**Comment by yutannihilation** [comment]
> In general I think it is safe to just substitute `match.fun()` with `as_function()`

Agreed.

**Comment by thomasp85** [comment]
@dkahle do you want to finish this off?

**Comment by dkahle** [comment]
@thomasp85 Yep! Is that what you're looking for?

File: R/stat-summary-2d.r [hunk 2]
```diff
@@ -100,7 +100,7 @@ StatSummary2d <- ggproto("StatSummary2d", Stat,
     xbin <- cut(data$x, xbreaks, include.lowest = TRUE, labels = FALSE)
     ybin <- cut(data$y, ybreaks, include.lowest = TRUE, labels = FALSE)
 
-    if (is.formula(fun)) fun <- rlang::as_function(fun)
+    if (is.formula(fun)) fun <- as_function(fun)
```
Comment (thomasp85) [inline R/stat-summary-2d.r]: I think you can just remove the `is.formula` check and call `as_function()` unconditionally... same with the other two places

**Comment by thomasp85** [comment]
Apart from the small comment you just need to fix the conflict in the news file, and it is good to go... thanks

**Comment by dkahle** [comment]
Hm. I pulled the news from upstream/master, updated it, and it still lists it as a conflict. What's the best way to resolve that? (Other than not wait two months before addressing your comments, that is.:)

**Comment by thomasp85** [comment]
You should merge the master branch into this branch (make sure master is current)

And don’t worry about time-to-respond. We all have shifting focus🙂

**Comment by thomasp85** [comment]
You should fix the merge conflict after merging in master. You still have the conflict in the NEWS file (the weird `<<<<< HEAD` notation)

Are you familiar with fixing merge conflicts or should I walk you through it?

**Comment by dkahle** [comment]
Just seeingx your second comment. I am familiar with merge conflicts; I thought I removed all merge conflict notes - did I miss something?

File: NEWS.md [hunk 3]
```diff
@@ -1,5 +1,19 @@
 # ggplot2 (development version)
 
+<<<<<<< HEAD
```
Comment (thomasp85) [inline NEWS.md]: There is something here🙂

Comment (dkahle) [reply NEWS.md]: Why yes, yes there is. 😂 I feel like this is the second time that my commit message should have a facepalm emoji.  

I actually searched the project for <<<<, but I realize now I was only searching through the R script files... Stay tuned. Thanks!

**Comment by thomasp85** [comment]
Thank you!

**Pull request merged**